src/lib/users/clients/inst_user_first.rb

```diff
@@ -0,0 +1,699 @@
+        "sha512" => _("SHA-512")
+      }
+
+      @import_checkbox = Left(
+        CheckBox(
+          Id(:import_ch),
+          # check box label
+          _("&Read User Data from a Previous Installation")
+        )
+      )
+
+      # button label
+      @import_button = PushButton(Id(:import), _("&Choose"))
+
+      @buttons = VBox(
+        VSpacing(0.5),
+
+        VBox(
+          Left(
+            RadioButton(
+              Id("users"),
+              Opt(:notify),
+              @button_label
+            )
+          ),
+          @import_available ?
+            HBox(
+              HSpacing(3),
+              @text_mode ?
+                VBox(@import_checkbox, Left(@import_button)) :
+                HBox(@import_checkbox, @import_button)
+            )
+          :
+            Empty()
+        ),
+
+        VSpacing(0.5)
+      )
+
+      @auth_term = VBox(
+        Frame(
+          _("Authentication Method"),
+          RadioButtonGroup(Id(:auth_method), @buttons)
+        )
+      )
+
+      # frame label
+      @encryption_term = Frame(
+        _("Password Encryption Type"),
+        RadioButtonGroup(
+          Id(:encryption_method),
+          VBox(
+            VSpacing(0.5),
+            # Radio button label: password encryption type
+            Left(RadioButton(Id("des"), _("&DES"))),
+            # Radio button label: password encryption type
+            Left(RadioButton(Id("md5"), _("&MD5"))),
+            # Radio button label: password encryption type
+            Left(RadioButton(Id("sha256"), _("SHA-&256"))),
+            # Radio button label: password encryption type
+            Left(RadioButton(Id("sha512"), _("SHA-&512"))),
+            VSpacing(0.5)
+          )
+        )
+      )
+
+      # help text for dialog "User Authentication Method" 1/2
+      @auth_help = _("<p><b>Authentication</b><br></p>") +
+        # help text for dialog "User Authentication Method" 2/2
+        _("<p>Select <b>Local</b> to authenticate users only by using the " +
+          "local files <i>/etc/passwd</i> and <i>/etc/shadow</i>.</p>")
+
+      # Help text for password expert dialog
+      @encryption_help = _("<p>Choose a password encryption method for local and system users.</p>") +
+        # Help text for password expert dialog
+        _("<p><b>SHA-512</b> is the current standard hash method. Using other " +
+          "algorithms is not recommended unless needed for compatibility purposes.</p>")
+
+      @users = UsersSimple.GetUsers
+      @user = {}
+
+      if @users.size > 1
+        @to_import = @users.map { |u| u["uid"] || "" }
+      elsif @users.size == 1
+        if @users.first["__imported"] != nil
+          @to_import = [@users.first.fetch("uid", "")]
+        else
+          @user = @users.first
+        end
+      end
+
+      @user_type = @user.fetch("type", "local")
+      @username = @user.fetch("uid", "")
+      @full_name = @user.fetch("cn", "")
+      @password = @user["userPassword"]
```

Comment:
this is a bit crazy. `@user` is set only if `@users.size == 1` and `@users.first["__imported"] == nil`, otherwise it is completely useless.